test/extended/federation.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# Runs the federation tests against a standard started server.
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+source "${OS_ROOT}/test/extended/setup.sh"
+
+os::test::extended::setup
+os::util::ensure::built_binary_exists kubefed
+os::test::extended::focus "$@"
+
+export OS_FEDERATION_NAME="${OS_FEDERATION_NAME:-origin-federation}"
+export OS_FEDERATION_NAMESPACE="${OS_FEDERATION_NAMESPACE:-federation-system}"
+export OS_FEDERATION_SERVICE_ACCOUNT="system:serviceaccount:${OS_FEDERATION_NAMESPACE}:default"
+
+os::test::junit::declare_suite_start "extended/federation"
+
+os::log::info "Deploying federation control plane"
+os::cmd::expect_success "kubefed init '${OS_FEDERATION_NAME}' --dns-provider=google-clouddns \
```

Comment:
when i follow this procedure against a `oc cluster up` environment, it is using the quay etcd image and its in a crash loop backoff with the following message:

```
$ oc logs origin-federation-apiserver-528877188-k3v5z --namespace=federation-system -c etcd
2017-06-08 05:38:01.129093 I | etcdmain: etcd Version: 3.1.7
2017-06-08 05:38:01.129264 I | etcdmain: Git SHA: 43b7507
2017-06-08 05:38:01.129269 I | etcdmain: Go Version: go1.7.5
2017-06-08 05:38:01.129272 I | etcdmain: Go OS/Arch: linux/amd64
2017-06-08 05:38:01.129277 I | etcdmain: setting maximum number of CPUs to 4, total number of available CPUs is 4
2017-06-08 05:38:01.129616 I | embed: listening for peers on http://localhost:2380
2017-06-08 05:38:01.129724 I | embed: listening for client requests on localhost:2379
2017-06-08 05:38:01.129791 C | etcdmain: cannot access data directory: mkdir /var/etcd/data: permission denied
```